src/services/EdFi.DataManagementService.Api/Core/Validation/EqualityConstraintValidator.cs

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Text.Json.Nodes;
+using EdFi.DataManagementService.Api.Core.Model;
+
+namespace EdFi.DataManagementService.Api.Core.Validation;
+
+public interface IEqualityConstraintValidator
+{
+    /// <summary>
+    /// Validates the equality constraints defined in MetaEd model are correct for the given API body.
+    /// </summary>
+    /// <param name="documentBody"></param>
+    /// <param name="equalityConstraints"></param>
+    /// <returns>Returns a list of validation failure messages.</returns>
+    IEnumerable<string>? Validate(JsonNode? documentBody, IEnumerable<EqualityConstraint> equalityConstraints);
+}
+
+public class EqualityConstraintValidator : IEqualityConstraintValidator
+{
+    public IEnumerable<string> Validate(JsonNode? documentBody, IEnumerable<EqualityConstraint> equalityConstraints)
+    {
+        var errors = new List<string>();
+        foreach (var equalityConstraint in equalityConstraints)
+        {
+
+            var sourcePath = Json.Path.JsonPath.Parse(equalityConstraint.SourceJsonPath.Value);
```

Comment:
It's an error in the ApiSchema.json file if the JsonPath is invalid. No need to do anything beyond crash explaining the ApiSchema.json file is bad.
The path may or may not exist. If it's not there, it's not a problem. We can assume it's optional because we've already JSON Schema validated. If it's an error in the ApiSchema.json, well, we can't detect that anyway.